std/algorithm/searching.d

```diff
@@ -1204,6 +1213,61 @@ bool endsWith(alias pred, R)(R doesThisEnd)
     }
 }
 
+/**
+Iterates the passed range and selects the extreme element with $(D less).
+If the extreme element occurs multiple time, the first occurrence will be
+returned.
+
+Params:
+    less = custom less predicate
+    r = Range from which the extreme value will be selected
+
+Returns: The extreme value according to $(D less) of the passed-in values.
+*/
+auto extremum(alias less, Range)(Range r)
+    if (isInputRange!Range && !isInfinite!Range)
+in
+{
+    assert(!r.empty, "r is an empty range");
+}
+body
+{
+    alias lessFun = binaryFun!less;
+    ElementType!Range extremeElement = r.front;
```

Comment:
I think `extremeElement` needs to be mutable even when `Range` is `const(T)[]`. Not sure what the typical solution is, maybe `std.traits.Unqual`.